### PR TITLE
Fix training loading

### PR DESCRIPTION
**Pull request by mattfred**
@frett @wrandall22 I created a test to for the training JSON parser and the training Dao and the test is not working. I was hoping you guys might see the problem. I parse in a JSON array which creates 2 training objects. I then save them, and retrieve them. When I retrieve them I only get one back. It seems like the Dao is updating when it should be inserting.

### CODE REVIEW

**Comment by wrandall22** [comment]
You're using lowercase `id` in the parser and uppercase `Id` in the test.

File: app/src/main/java/com/expidev/gcmapp/json/TrainingJsonParser.java [hunk 1]
```diff
@@ -32,15 +33,23 @@
     @NonNull
     public static Training parseTraining(@NonNull final JSONObject json) throws JSONException {
         final Training training = new Training();
-        training.setId(json.getInt("id"));
+        if (json.has("id")) training.setId(json.getInt("id"));
```
Comment (frett) [inline app/src/main/java/com/expidev/gcmapp/json/TrainingJsonParser.java]: you may want to use `json.optInt("Id")` instead.

Something else I just noticed is that the id is currently an int, in order for it to be used by internal android functions it should be a long.

**Comment by mattfred** [comment]
Ok thanks for your help. This is working now.

**Comment by mattfred** [comment]
@frett - I decided to wait on building the training drill down since I figured I could reuse some of your code from the drill down you are working on.

**Comment by frett** [comment]
looks good to me

**Comment by frett** [comment]
@mattfred sounds good, I have my current state up in the churches branch. I have it driving up the dialog already and am working on tracking the dirty state of a church in the db since it can be updated offline and needs to sync when it has internet.

**Comment by mattfred** [comment]
So this is a side question, but since the app is supposed to work offline, how is the map going to load? This seems to be a pretty important part of the app.

**Comment by frett** [comment]
that's a very good point, I'm assuming google maps caches map data for offline usage. I'm sure I'll notice quite a few offline things while in South Africa the next couple weeks.

**Comment by wrandall22** [comment]
I tried earlier in airplane mode and the map came up okay, not sure if you can move around or zoom, though.

**Comment by frett** [comment]
I just did airplane mode and was able to navigate around the map. zooming in didn't load more detailed maps unless I had previously zoomed in there.

**Pull request merged**